server/src/main/java/org/opensearch/search/aggregations/support/MissingValues.java

```diff
@@ -359,7 +360,8 @@ public long getValueCount() {
 
             @Override
             public int docValueCount() {
-                return Math.max(1, values.docValueCount());
+                // If we don't have ordinals, then we just have the missing value
+                return hasOrds ? values.docValueCount() : 1;
             }
```

Comment:
The `nextOrd` just below this is also not checking for docValueCount.

Reply:
That's okay. If the caller calls `nextOrd()` too many times, the behavior is undefined.